src/aggregator/aggregator/generic_elem.go

```diff
@@ -320,7 +319,9 @@ func (e *GenericElem) expireValuesWithLock(
 			// if this current value is closed and clean it will no longer be flushed. this means it's safe
 			// to remove the previous value since it will no longer be needed for binary transformations. when the
 			// next value is eligible to be expired, this current value will actually be removed.
-			if prevV, ok := e.values[e.minStartTime]; ok {
+			// if we're currently pointing at the start skip this there is no previous for the start. this ensures
+			// we always keep at least one value in the map for binary transformations.
+			if prevV, ok := e.values[e.minStartTime]; ok && currStart != e.minStartTime {
```

Comment:
Ah ok this makes sense. nit: this comment wording I think is missing a "because"